pkg/core/resources/apis/mesh/dataplane_overview.go

```diff
@@ -98,14 +102,14 @@ func NewDataplaneOverviews(dataplanes DataplaneResourceList, insights DataplaneI
 	for _, dataplane := range dataplanes.Items {
 		overview := DataplaneOverviewResource{
 			Meta: dataplane.Meta,
-			Spec: mesh_proto.DataplaneOverview{
-				Dataplane:        proto.Clone(&dataplane.Spec).(*mesh_proto.Dataplane),
+			Spec: &mesh_proto.DataplaneOverview{
+				Dataplane:        dataplane.Spec,
```

Comment:
OK, so this is just an overview, so I guess Spec is immutable here, but how do we guarantee safety now that we use pointers everywhere?

Reply:
Yeah... we cannot really :( that's the huge downside of having pointers.

Here I noticed that the cloning is unnecessary so I removed it.